src/main/java/com/hivemq/mqtt/handler/connect/ConnectHandler.java

```diff
@@ -651,60 +651,58 @@ private void sendConnackSuccess(final @NotNull ChannelHandlerContext ctx, final
     private @NotNull ListenableFuture<Void> disconnectClientWithSameClientId(
             final @NotNull CONNECT msg, final @NotNull ClientConnection clientConnection, final int retry) {
 
-        final Lock lock = stripedLock.get(msg.getClientIdentifier());
-        lock.lock();
```

Comment:
Is it really safe to remove thise lock?

Reply:
Yes, as the tryPersist is safe to be called concurrently.

Reply:
But the lock guarded a lot more than just the tryPersist before.

Reply:
The remaining code in the previous lock is also protected by the ClientState and the disconnect being processed in the event loop.